### ISSUE TITLE
DOTS collections preview package is not compatible

### ISSUE THREAD

**Issue by BrundleThwaite** [Unity-Technologies/com.unity.demoteam.hair#36]
Currently the hair package breaks when using the preview version of dots. This appears to be because of changes to the collections package for example "UnsafeParallelHashSet" and "NativeArray" tend to give problems.

Is there any plan to make these compatible with each other?

**Comment by fuglsang** [comment]
Hi there --
With which version of Unity and the collections package are you seeing errors?
Also, please make sure to pull the latest version of the hair package. (E.g. the rename of UnsafeHashSet->UnsafeParallelHashSet was accounted for in 555836c)

**Comment by BrundleThwaite** [comment]
Here are a couple of screen grabs. The Unity version is 2022.1f1.
It is a standard new project.
I then installed HDRP and updated it using the wizard.
I installed the hair using the url: https://github.com/Unity-Technologies/com.unity.demoteam.hair.git
I updated the package.
I added the sample scene.
I installed Entities Graphics. The Project broke, the mesh instance reference breaks and I get the the error in the console.
Updating collections to 2.1.0-pre.6 and the error remains.

The first screen shot shows the packages I have installed.
The second is the error location in Visual Studio. 

![image](https://user-images.githubusercontent.com/18170954/210284527-f2c43fe4-e5d4-4803-9bd1-aed221c3706c.png)
![image](https://user-images.githubusercontent.com/18170954/210284557-740f9e38-0af3-4563-80c9-4802f6158130.png)
